src/System.Runtime/tests/System/HashCodeTests.netcoreapp.cs

```diff
@@ -0,0 +1,227 @@
+            HashCode.Combine(1, 2, 3, 4, 5),
+            HashCode.Combine(1, 2, 3, 4, 5, 6),
+            HashCode.Combine(1, 2, 3, 4, 5, 6, 7),
+            HashCode.Combine(1, 2, 3, 4, 5, 6, 7, 8),
+
+            HashCode.Combine(2),
+            HashCode.Combine(2, 3),
+            HashCode.Combine(2, 3, 4),
+            HashCode.Combine(2, 3, 4, 5),
+            HashCode.Combine(2, 3, 4, 5, 6),
+            HashCode.Combine(2, 3, 4, 5, 6, 7),
+            HashCode.Combine(2, 3, 4, 5, 6, 7, 8),
+            HashCode.Combine(2, 3, 4, 5, 6, 7, 8, 9),
+        };
+
+        for (int i = 0; i < hcs.Length; i++)
+            for (int j = 0; j < hcs.Length; j++)
+            {
+                if (i == j) continue;
+                Assert.NotEqual(hcs[i], hcs[j]);
+            }
+    }
+
+    [Fact]
+    public static void HashCode_Combine_Add_1()
+    {
+        var hc = new HashCode();
+        hc.Add(1);
+        Assert.Equal(hc.ToHashCode(), HashCode.Combine(1));
+    }
+
+    [Fact]
+    public static void HashCode_Combine_Add_2()
+    {
+        var hc = new HashCode();
+        hc.Add(1);
+        hc.Add(2);
+        Assert.Equal(hc.ToHashCode(), HashCode.Combine(1, 2));
+    }
+
+    [Fact]
+    public static void HashCode_Combine_Add_3()
+    {
+        var hc = new HashCode();
+        hc.Add(1);
+        hc.Add(2);
+        hc.Add(3);
+        Assert.Equal(hc.ToHashCode(), HashCode.Combine(1, 2, 3));
+    }
+
+    [Fact]
+    public static void HashCode_Combine_Add_4()
+    {
+        var hc = new HashCode();
+        hc.Add(1);
+        hc.Add(2);
+        hc.Add(3);
+        hc.Add(4);
+        Assert.Equal(hc.ToHashCode(), HashCode.Combine(1, 2, 3, 4));
+    }
+
+    [Fact]
+    public static void HashCode_Combine_Add_5()
+    {
+        var hc = new HashCode();
+        hc.Add(1);
+        hc.Add(2);
+        hc.Add(3);
+        hc.Add(4);
+        hc.Add(5);
+        Assert.Equal(hc.ToHashCode(), HashCode.Combine(1, 2, 3, 4, 5));
+    }
+
+    [Fact]
+    public static void HashCode_Combine_Add_6()
+    {
+        var hc = new HashCode();
+        hc.Add(1);
+        hc.Add(2);
+        hc.Add(3);
+        hc.Add(4);
+        hc.Add(5);
+        hc.Add(6);
+        Assert.Equal(hc.ToHashCode(), HashCode.Combine(1, 2, 3, 4, 5, 6));
+    }
+
+    [Fact]
+    public static void HashCode_Combine_Add_7()
+    {
+        var hc = new HashCode();
+        hc.Add(1);
+        hc.Add(2);
+        hc.Add(3);
+        hc.Add(4);
+        hc.Add(5);
+        hc.Add(6);
+        hc.Add(7);
+        Assert.Equal(hc.ToHashCode(), HashCode.Combine(1, 2, 3, 4, 5, 6, 7));
+    }
+
+    [Fact]
+    public static void HashCode_Combine_Add_8()
+    {
+        var hc = new HashCode();
+        hc.Add(1);
+        hc.Add(2);
+        hc.Add(3);
+        hc.Add(4);
+        hc.Add(5);
+        hc.Add(6);
+        hc.Add(7);
+        hc.Add(8);
+        Assert.Equal(hc.ToHashCode(), HashCode.Combine(1, 2, 3, 4, 5, 6, 7, 8));
+    }
+
```

Comment:
If I recall correctly, adding any non-null object `o` and adding `o.GetHashCode()` (or the deferred-to-an-assistant equivalent) should be the same.  If true, tests for that seem good.  (If false... tests for that seem not unreasonable).